contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3752,3 +3752,73 @@ exec_stmt_grantschema(PLtsql_execstate *estate, PLtsql_stmt_grantschema *stmt)
 	}
 	return PLTSQL_RC_OK;
 }
+
+/*
+ * ALTER AUTHORIZATION ON DATABASE::dbname TO loginname
+ */
+static int
+exec_stmt_change_dbowner(PLtsql_execstate *estate, PLtsql_stmt_change_dbowner *stmt)
+{
+	bool granting_to_self = false;
+	
+	/* Verify target database exists. */
+	if (!DbidIsValid(get_db_id(stmt->db_name)))
+	{
+		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_DATABASE),	
+						errmsg("Cannot find the database '%s', because it does not exist or you do not have permission.", stmt->db_name)));
+	}
+
+	/* Verify new owner exists as a login. */
+	if (get_role_oid(stmt->new_owner_name, true) == InvalidOid)
+	{
+		ereport(ERROR, (errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+						errmsg("Cannot find the principal '%s', because it does not exist or you do not have permission.", stmt->new_owner_name)));
+	}
+	
+	/* SQL Server allows granting ownership to yourself when you are owner already, even without having sysadmin role. */
+	if (get_role_oid(stmt->new_owner_name, true) == GetSessionUserId())  // Granting ownership to myself?
+	{
+		/* Is the current login already DB owner? */
+		if (get_role_oid(get_owner_of_db(stmt->db_name), true) == GetSessionUserId())
+		{
+			granting_to_self = true;		
```

Comment:
We can get rid of this variable  `granting_to_self` and directly add the following lines here.
```
update_db_owner(stmt->new_owner_name, stmt->db_name);
return PLTSQL_RC_OK;
```

Reply:
done